core/lib/dal/src/base_token_dal.rs

```diff
@@ -0,0 +1,63 @@
+use std::num::NonZeroU64;
+
+use bigdecimal::{BigDecimal, FromPrimitive};
+use zksync_db_connection::{connection::Connection, error::DalResult, instrument::InstrumentExt};
+use zksync_types::base_token_ratio::BaseTokenRatio;
+
+use crate::{models::storage_base_token_ratio::StorageBaseTokenRatio, Core};
+
+#[derive(Debug)]
+pub struct BaseTokenDal<'a, 'c> {
+    pub(crate) storage: &'a mut Connection<'c, Core>,
+}
+
+impl BaseTokenDal<'_, '_> {
+    pub async fn insert_token_ratio(
+        &mut self,
+        numerator: NonZeroU64,
+        denominator: NonZeroU64,
+        ratio_timestamp: &chrono::NaiveDateTime,
+    ) -> DalResult<usize> {
+        let row = sqlx::query!(
+            r#"
+            INSERT INTO
+                base_token_ratios (numerator, denominator, ratio_timestamp, created_at, updated_at)
+            VALUES
+                ($1, $2, $3, NOW(), NOW())
+            RETURNING
+                id
+            "#,
+            BigDecimal::from_u64(numerator.get()),
+            BigDecimal::from_u64(denominator.get()),
+            ratio_timestamp,
+        )
+        .instrument("insert_token_ratio")
+        .fetch_one(self.storage)
+        .await?;
+
+        Ok(row.id as usize)
+    }
+
+    // TODO (PE-128): pub async fn mark_l1_update()
```

Comment:
Nit: It's better to leave TODO comments on something that is already implemented but misses some functionality. If nothing has been done yet, it's better to just rely on the task; otherwise such comments look confusing.

Reply:
Agreed as a general principle, but the purpose of this here, and other TODOs around this PR is to not only place the TODO but also indicate to the reader the intended full flow.

Reply:
The problem with such TODOs is that pretty often people either forget to remove them or priorities change and these TODOs are left hanging. I think the intended full flow can be implied from the already existing implementation, so idk.